### ISSUE TITLE
(aws-apigateway): No Deployment for imported RestAPI after adding resources or methods

### ISSUE THREAD

**Issue by Adiotastic** [aws/aws-cdk#12589]
### The Question
I have created an api gateway like:

`const restAPI = newRestApi(this, 'APIGateway', {
      deploy: true,
      deployOptions: {
        stageName: 'dev',
        description: 'dev stage',
      },
    });`

In a different stack I want to use the gateway to add new resources and methods from lambda functions, like deploying different separated lambda applications to the same gateway.

I imported the RestApi like this:
` const restApi = RestApi.fromRestApiAttributes(this, 'RestApi', { ... }); `

I add a new resource and method for the lambda I want to add as a gateway endpoint:
`const integration =  restApi.root.addResource('test').addMethod('GET', new LambdaIntegration(testLambda));`

**Issue**
In the aws console I can see the new resource 'test' as well as the new method, but there is no new deployment in the DEV Stage.
MIght this be in issue with the CDK or am I doing this wrong?

### Environment

  - **CDK CLI Version:** 1.85
  - **Module Version:** 1.85
  - **Node.js Version:** 10.14.1
  - **OS:** macOS Catalina
  - **Language (Version):** TypeScript

**Comment by achamwada** [comment]
Hi @Adiotastic  , 

Have you tried to set  deploy  to false remove deployOptions so you can manually create your deployment for your stages 

`deploy: true` creates a deployment to a `prod` stage 

```
const method = restApi.root.addResource('test').addMethod('GET', new LambdaIntegration(testLambda));

const deployment = new Deployment(this, 'api-gateway-deployment', {
        api: restApi,
        retainDeployments: true,
      });

deployment.node.addDependency(method);

 new Stage(this, `${stageName}_stage`, {
        deployment,
        stageName,
      });
```

**Comment by Adiotastic** [comment]
Hi @achamwada ,
i tried that once, but cannot remember if I used "addDependency" for this.
Since this was a blocker, I solved it this way:

I extracted the creation of new resources and methods to the gateway stack and generated the arn for the Integration Lambdas there. This was the way a lot of other guys handle problems like this.

Even if this is more like strong coupling between the lambda names/arn and the integration, and also not the way i would prefer to go, it guarantees that I can add multiple separated lambda applications to the same gateway.

Thank you anyway. If I have some time to spend, I will try to implement your idea to check if it works.
